cedar-policy-validator/src/schema/namespace_def.rs

```diff
@@ -248,81 +409,159 @@ impl EntityTypesDef {
+/// undeclared entity/common types. Furthermore, entity/common type references
+/// in `attributes` may or may not be fully qualified yet, depending on `N`.
 #[derive(Debug)]
-pub struct EntityTypeFragment {
-    /// The attributes record type for this entity type. The type is wrapped in
-    /// a `WithUnresolvedTypeDefs` because it may refer to common types which
-    /// have not yet been resolved/inlined (e.g., because they are not defined
-    /// in this schema fragment).
-    pub(super) attributes: WithUnresolvedTypeDefs<Type>,
+pub struct EntityTypeFragment<N> {
+    /// The attributes record type for this entity type. This may contain
+    /// references to common types which have not yet been resolved/inlined
+    /// (e.g., because they are not defined in this schema fragment).
+    pub(super) attributes: SchemaType<N>,
     /// Direct parent entity types for this entity type.
-    /// These are fully qualified entity types, but may be entity types declared
-    /// in a different namespace or schema fragment.
+    /// These entity types may be declared in a different namespace or schema
+    /// fragment.
     /// We will check for undeclared parent types when combining fragments into
     /// a [`crate::ValidatorSchema`].
-    pub(super) parents: HashSet<EntityType>,
+    pub(super) parents: HashSet<N>,
 }
 
-impl EntityTypeFragment {
-    /// Construct an [`EntityTypeFragment`] by converting the structures used by
-    /// the schema format to those used internally by the validator.
+impl EntityTypeFragment<ConditionalName> {
+    /// Construct a [`EntityTypeFragment<ConditionalName>`] by converting the
+    /// structures used by the schema format to those used internally by the
+    /// validator.
     pub(crate) fn from_raw_entity_type(
-        entity_type: schema_file_format::EntityType<RawName>,
+        schema_file_type: schema_file_format::EntityType<RawName>,
         schema_namespace: Option<&Name>,
-        extensions: Extensions<'_>,
-    ) -> Result<Self> {
-        Ok(Self {
-            attributes: try_schema_type_into_validator_type(
-                entity_type
-                    .shape
-                    .into_inner()
-                    .qualify_type_references(schema_namespace),
-                extensions,
-            )?,
-            parents: entity_type
+    ) -> Self {
+        Self {
+            attributes: schema_file_type
+                .shape
+                .into_inner()
+                .conditionally_qualify_type_references(schema_namespace),
+            parents: schema_file_type
                 .member_of_types
                 .into_iter()
-                .map(|raw_name| raw_name.qualify_with(schema_namespace).into())
+                .map(|raw_name| {
+                    raw_name.conditionally_qualify_with(schema_namespace, ReferenceType::Entity)
+                }) // REVIEW: here's another place to adjust if we decide we want to allow common types in this position
```

Comment:
I think there is some value in defining a common type that points to an entity type.

Simple example (there may be more):
Suppose I have a namespace or fragment that uses `User` a lot, and I want to be able to flip between two different definitions of `User` for testing, or maybe because I’m in the process of migrating, or something.  Then I can change between `type User = User1;` and `type User = User2;` without changing `User` everywhere it appears.

Reply:
At any rate, this was punted to #1064; we can continue this discussion there

Reply:
Edited this comment to match the others that point to 1064